Bottom: 65bd216d686ec7fbe69f7ef857b0abdd61fa62fb
Top:    b7cadde8a9fbda44443eb4b6137a961f9d4d607c
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-03-01 10:39:50 +0800

Refresh of virtio-net-re-enable

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 771a7d8..2de97b6 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1048,7 +1048,7 @@ static int add_recvbuf_mergeable(struct virtnet_info *vi,
 	struct page_frag *alloc_frag = &rq->alloc_frag;
 	unsigned int headroom = virtnet_get_headroom(vi);
 	unsigned int tailroom = headroom ? sizeof(struct skb_shared_info) : 0;
-	unsigned int room = headroom + tailroom;
+	unsigned int room = SKB_DATA_ALIGN(headroom + tailroom);
 	char *buf;
 	void *ctx;
 	int err;
@@ -2609,7 +2609,7 @@ static ssize_t mergeable_rx_buffer_size_show(struct netdev_rx_queue *queue,
 	avg = &vi->rq[queue_index].mrg_avg_pkt_len;
 	return sprintf(buf, "%u\n",
 		       get_mergeable_buf_len(&vi->rq[queue_index], avg,
-				             headroom + tailroom));
+				       SKB_DATA_ALIGN(headroom + tailroom)));
 }
 
 static struct rx_queue_attribute mergeable_rx_buffer_size_attribute =
